nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,730 @@
+
+    def apply_affine(self, affine, lazy=False):
+        """ Applies an affine transformation on the points of each streamline.
+
+        If `lazy` is not specified, this is performed *in-place*.
+
+        Parameters
+        ----------
+        affine : ndarray of shape (4, 4)
+            Transformation that will be applied to every streamline.
+        lazy : {False, True}, optional
+            If True, streamlines are *not* transformed in-place and a
+            :class:`LazyTractogram` object is returned. Otherwise, streamlines
+            are modified in-place.
+
+        Returns
+        -------
+        tractogram : :class:`Tractogram` or :class:`LazyTractogram` object
+            Tractogram where the streamlines have been transformed according
+            to the given affine transformation. If the `lazy` option is true,
+            it returns a :class:`LazyTractogram` object, otherwise it returns a
+            reference to this :class:`Tractogram` object with updated
+            streamlines.
+        """
+        if lazy:
+            lazy_tractogram = LazyTractogram.from_tractogram(self)
+            return lazy_tractogram.apply_affine(affine)
+
+        if len(self.streamlines) == 0:
+            return self
+
+        BUFFER_SIZE = 10000000  # About 128 Mb since pts shape is 3.
+        for start in range(0, len(self.streamlines.data), BUFFER_SIZE):
+            end = start + BUFFER_SIZE
+            pts = self.streamlines._data[start:end]
+            self.streamlines.data[start:end] = apply_affine(affine, pts)
+
+        if self.affine_to_rasmm is not None:
+            # Update the affine that brings back the streamlines to RASmm.
+            self.affine_to_rasmm = np.dot(self.affine_to_rasmm,
+                                          np.linalg.inv(affine))
+
+        return self
+
+    def to_world(self, lazy=False):
+        """ Brings the streamlines to world space (i.e. RAS+ and mm).
+
+        If `lazy` is not specified, this is performed *in-place*.
+
+        Parameters
+        ----------
+        lazy : {False, True}, optional
+            If True, streamlines are *not* transformed in-place and a
+            :class:`LazyTractogram` object is returned. Otherwise, streamlines
+            are modified in-place.
+
+        Returns
+        -------
+        tractogram : :class:`Tractogram` or :class:`LazyTractogram` object
+            Tractogram where the streamlines have been sent to world space.
+            If the `lazy` option is true, it returns a :class:`LazyTractogram`
+            object, otherwise it returns a reference to this
+            :class:`Tractogram` object with updated streamlines.
+        """
+        if self.affine_to_rasmm is None:
+            msg = ("Streamlines are in a unknown space. This error can be"
+                   " avoided by setting the 'affine_to_rasmm' property.")
+            raise ValueError(msg)
+
+        return self.apply_affine(self.affine_to_rasmm, lazy=lazy)
```

Comment:
Hmm, I have second thoughts. I cannot put the check in `to_world` because of the `lazy` option of the method, so it would have to go back in the `apply_affine` method :P.